hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileWriter.java

```diff
@@ -135,14 +132,16 @@ private StoreFileWriter(FileSystem fs, Path path, final Configuration conf, Cach
       // init bloom context
       switch (bloomType) {
         case ROW:
-          bloomContext = new RowBloomContext(generalBloomFilterWriter, comparator);
+          bloomContext =
+            new RowBloomContext(generalBloomFilterWriter, fileContext.getCellComparator());
           break;
         case ROWCOL:
-          bloomContext = new RowColBloomContext(generalBloomFilterWriter, comparator);
+          bloomContext =
+            new RowColBloomContext(generalBloomFilterWriter, fileContext.getCellComparator());
```

Comment:
So every where it comes from the context only including for the blooms

Reply:
Yes sir. One place only instead of hardcodings and guesses at which to use.